Userland/Applications/PixelPaint/Filters/Filter.h

```diff
@@ -24,15 +24,16 @@ class Filter {
 
     virtual ~Filter() {};
 
-    Filter(ImageEditor* editor)
-        : m_editor(editor) {};
+    Filter(ImageEditor* editor);
 
     Function<void(void)> on_settings_change;
 
 protected:
     ImageEditor* m_editor { nullptr };
     RefPtr<GUI::Widget> m_settings_widget { nullptr };
     void update_preview();
+
+    RefPtr<Core::Timer> m_update_timer;
```

Comment:
I think it would be cleaner to have a NonnullRefPtr here and initialize it with a member initialization thingy in the constructor directly.